core/include/kernel/dt_driver.h

```diff
@@ -0,0 +1,116 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2021, Linaro Limited
+ * Copyright (c) 2021, Bootlin
+ */
+
+#ifndef __DT_DRIVER_H
+#define __DT_DRIVER_H
+
+#include <kernel/dt.h>
+#include <stdint.h>
+#include <sys/queue.h>
+#include <tee_api_types.h>
+
+/* Opaque reference to DT driver device provider instance */
+struct dt_driver_provider;
+
+/**
+ * struct dt_driver_phandle_args - Devicetree phandle arguments
+ * @args_count: Count of cells for the device
+ * @args: Device consumer specifiers
+ */
+struct dt_driver_phandle_args {
+	int args_count;
+	uint32_t args[];
+};
+
+/*
+ * get_of_device_func - Callback function for returning a driver private
+ *	instance based on a FDT phandle with possible arguments and the
+ *	registered dt_driver private data reference.
+ *
+ * @parg: phandle argument(s) referencing the device in the FDT.
+ * @data: driver private data registered in struct dt_driver.
+ *
+ * Return a device opaque reference, e.g. a struct clk pointer for a clock
+ * driver, or NULL if not found.
+ */
+typedef void *(*get_of_device_func)(struct dt_driver_phandle_args *parg,
+				    void *data);
+
+/**
+ * dt_driver_register_provider - Register a driver provider
+ *
+ * @fdt: Device tree to work on
+ * @nodeoffset: Node offset in the FDT
+ * @get_of_device: Function to match the devicetree with a device instance
+ * @data: Data which will be passed to the @get_of_device callback
+ * @type: Driver type
+ *
+ * @get_of_device returns a void *. Driver provider is expected to
+ * include a shim helper to cast to device reference into provider driver
+ * target structure reference (e.g (struct clk *) for clock devices).
+ */
+TEE_Result dt_driver_register_provider(const void *fdt, int nodeoffset,
+				       get_of_device_func get_of_device,
+				       void *data, enum dt_driver_type type);
+
+/*
+ * Return a device instance based on a property name and FDT information
```

Comment:
Somtime the function name is present, sometimes not. Maybe add function name here ?

Reply:
right, i'll add it where there is a detailed inline description of the function (the prototypes you pointed out).